### PR TITLE
[ICD] Add Keep active during MDNS resolution ICD Check-In Sender

### LINKED ISSUE #30492: [1.3][ICD] Stay active during MDNS resolution for sending checking messages
When resolving NodeID, the device needs to stay active.

### PR DESCRIPTION
**Pull request by jepenven-silabs**
Add Keep Active during MDNS resolution

Fix #30492

Breakout of PR #30526 to speed up review and merge.

Also remove Message layer dependency on the ICDManager.

### CODE REVIEW

File: src/app/icd/ICDManager.h [hunk 1]
```diff
@@ -128,6 +129,7 @@ class ICDManager : public ICDListener
     static void OnTransitionToIdle(System::Layer * aLayer, void * appState);
 
     static uint8_t OpenExchangeContextCount;
+    static uint8_t CheckInRequestCount;
```
Comment (mkardous-silabs) [inline src/app/icd/ICDManager.h:132]:
```suggestion
   uint8_t mOpenExchangeContextCount;
   uint8_t mCheckInRequestCount;
```

These don't need to be static anymore.

File: src/app/icd/ICDManager.cpp [hunk 2]
```diff
@@ -41,6 +41,7 @@ using namespace chip::app::Clusters;
 using namespace chip::app::Clusters::IcdManagement;
 
 uint8_t ICDManager::OpenExchangeContextCount = 0;
+uint8_t ICDManager::CheckInRequestCount      = 0;
```
Comment (mkardous-silabs) [inline src/app/icd/ICDManager.cpp:44]:
```suggestion
```

File: src/app/icd/ICDManager.h (hunk 1 above)
Comment (jepenven-silabs) [reply src/app/icd/ICDManager.h:132]: Pretty sure @jmartinez-silabs had a reason to put that static.

**Review by mkardous-silabs — APPROVED** [review]
- [x] LGTM after the last comment

**Review by jmartinez-silabs — APPROVED** [review]

File: src/app/icd/ICDManager.h (hunk 1 above)
Comment (jmartinez-silabs) [reply src/app/icd/ICDManager.h]: Yes my bad, When updating from ICDEventManager mechanic to the ICDNotifer, the static functions became just public (non-static), I forgot to move this static variable to the object too

**Review by bzbarsky-apple — APPROVED** [review]

**Review by mkardous-silabs — APPROVED** [review]

**Review by andy31415 — APPROVED** [review]

**Pull request merged**